### PR TITLE
Add v3 signature to a packet

### PR DESCRIPTION
**Pull request by flicker581**
V3 protocol require packet signatures before CRC and "whitening".

Current LampSmart Pro does not support sending commands without this signature, so, I suppose it is harmless.

### CODE REVIEW

**Comment by oxcid3** [comment]
Watching this and testing on my light. Not currently working but very excited that you're trying to sort. Thanks 🤞🏻

**Review by aronsky — CHANGES_REQUESTED** [review]
A couple of notes:

1. We'll need to verify that changing that field from 0 to the signature value does not break lights that don't use the signature (e.g., the lights I use :) ). I can help with that.
2. If 1 fails, we'll need to make the changes in this PR optional, i.e. - make the use of the signature configurable.
3. I don't want to add an AES implementation to the source tree unless absolutely necessary (it doesn't even have a license at the top - excuse my bluntness, but is this code yours?). I prefer to use libraries where possible. It seems that ESP32 supports hardware AES encryption/decryption - why not use it? See an example here: https://github.com/ffosilva/AES32.

File: components/lampsmart_pro_light/lampsmart_pro_light.cpp [hunk 1]
```diff
@@ -167,11 +169,25 @@ void LampSmartProLight::send_packet(uint16_t cmd, uint8_t cold, uint8_t warm) {
...
 
+  uint8_t sigkey[16] = {0, 0, 0, 0x0D, 0xBF, 0xE6, 0x42, 0x68, 0x41, 0x99, 0x2D, 0x0F, 0xB0, 0x54, 0xBB, 0x16};
+  sigkey[0] = seed & 0xff;
+  sigkey[1] = (seed >> 8) & 0xff;
+  sigkey[2] = this->tx_count_;
+  struct AES_ctx aes_ctx;
+  AES_init_ctx(&aes_ctx, sigkey);
+  uint8_t aes_buf[16];
+  memcpy(aes_buf, &packet.raw[8], 16);  
+  AES_ECB_encrypt(&aes_ctx, aes_buf);
```
Comment (aronsky) [inline components/lampsmart_pro_light/lampsmart_pro_light.cpp:185]: This needs to be in a separate function.

**Comment by flicker581** [comment]
1. LampSmartPro 1.2.9 sends only v3 messages (i.e. with the signature). This makes me feel that the change is backward compatible. I have no ways to test, because I do not own any of these lamps. I have checked only that:
- the signature generated by the code is passing the same check as LampSmartPro's signature.
- my friend's lamp is working with this code. 
2. Ok. Unfortunately, base code have no provisions for pluggable algorithms.
3. AES code is from https://github.com/kokke/tiny-AES-c, where it is published as "public domain". Now I have replaced it with mbedtls, which comes in arduino-espressif32. Have no idea how stable this dependency is.

Feel free to use my code as you wish, integrate it some way or drop it. I personally have no special interest in it. Sorry.

@oxcid3 
So, if this does not work for you (and you are sure doing thing right), your lamp needs older algo. Tried lampify yet?

**Comment by aronsky** [comment]
@flicker581 - thanks for the changes! I'll test it as soon as I get the chance (to verify that it's backward-compatible), and get back to you.

**Review by aronsky — APPROVED** [review]

**Comment by aronsky** [comment]
Thanks for the changes. I didn't get a chance to test them out (as I switched to a different LED driver that works with a different app). I'm merging this - hopefully it works for everyone. In any case, for now, the version without the signature remains available in the refactored code - and in the future, I'll add support to make the signing optional.

**Pull request merged**